src/NUnitFramework/framework/Internal/Execution/EventQueue.cs

```diff
@@ -15,13 +29,20 @@ namespace NUnit.Framework.Internal.Execution
     /// or to queue them for forwarding on another thread or at
     /// a later time.
     /// </summary>
-    public abstract class Event
+    public abstract class Event : IEvent<ITestListener>, IDisposable
     {
         /// <summary>
         /// The Send method is implemented by derived classes to send the event to the specified listener.
         /// </summary>
         /// <param name="listener">The listener.</param>
         public abstract void Send(ITestListener listener);
+
+        /// <summary>
+        /// Not quite sure why this is needed, but the compiler complained.
+        /// </summary>
+        public void Dispose()
```

Comment:
If this is not `virtual` derived classes cannot implement it if needed.

Reply:
good point! Fixed.